### PR TITLE
capabilities_parser parse ast and json

### PR DESCRIPTION
**Pull request by piotrkochan**
* parse json - .json files
* parse ast - raw .py dist
* parse .py as normal if other ways failed

### CODE REVIEW

File: seleniumbase/core/capabilities_parser.py [hunk 1]
```diff
@@ -169,11 +171,27 @@ def get_desired_capabilities(cap_file):
...
-            num_capabilities += 1
+            capabilities[key] = value
             continue
 
-    if num_capabilities == 0:
+    return capabilities
+
+
+def get_desired_capabilities(cap_file):
+    f = open(cap_file, 'r')
```
Comment (mdmintz) [inline seleniumbase/core/capabilities_parser.py]: You have to verify the file type before doing any kind of file parsing.

**Comment by mdmintz** [comment]
@piotrkochan I added the review. You should verify the file type before parsing the desired capabilities file.

File: seleniumbase/core/capabilities_parser.py [hunk 2]
```diff
@@ -169,11 +171,27 @@ def get_desired_capabilities(cap_file):
...
+    f.close()
+
+    if cap_file.endswith('.py'):
+        capabilities = _parse_ast(all_code)
+        if not capabilities:
+            capabilities = _parse_manual(all_code)
+    elif cap_file.endswith('.json'):
+        capabilities = json.loads(all_code)
+    else:
+        raise Exception("\n\n`%s` is not a Python or JSON file!\n" % cap_file)
```
Comment (mdmintz) [inline seleniumbase/core/capabilities_parser.py]: This part should happen before ``f = open(cap_file, 'r')`` if the file type is invalid.

Comment (piotrkochan) [reply seleniumbase/core/capabilities_parser.py]: fixed

**Comment by mdmintz** [comment]
@piotrkochan Merged! It'll be part of the next release. Thank you!

**Pull request merged**